src/main/java/ladder/LadderApplication.java

```diff
@@ -1,18 +1,26 @@
 package ladder;
 
-import ladder.domain.LadderFactory;
+import ladder.domain.Ladder;
+import ladder.domain.Results;
 import ladder.view.InputView;
 import ladder.view.OutputView;
 
 
 public class LadderApplication {
 
+
     public static void main(String[] args) {
         InputView inputView = new InputView();
         OutputView outputView = new OutputView();
 
-        LadderFactory ladderFactory = new LadderFactory(inputView.inputNames(), inputView.inputHeight());
-        outputView.printNames(ladderFactory.getNames());
-        outputView.printLadder(ladderFactory.getLadder());
+        Ladder ladder = new Ladder(inputView.inputNames(), inputView.inputPrizes(), inputView.inputHeight());
+        outputView.printLadder(ladder);
+
+        Results results = ladder.getResults();
+        String name;
+        while ((name = inputView.inputNameForResult()) != null){
+            outputView.printResult(results, name);
+        }
+        outputView.printAllResult(results);
```

Comment:
null을 제어 흐름에 사용하는 것은 알맞지 않은 것 같아요!
명확한 방법으로 흐름을 제어하는 것이 더 좋겠습니다.


심플하게 작성해 본다면 아래와 같은 코드가 나올 것 같네요

코드는 참고만 해주세요~
```suggestion
        while (true) {
            String name = inputView.inputNameForResult();
            if ("all".equals(name)) {
                break;
            }
            outputView.printResult(results, name);
        }
        outputView.printAllResult(results);
```

depth가 신경쓰인다면 메소드 추출을 통해서도 해결할 수 있을 것 같습니다!